addons/mail/static/tests/mail_utils.test.js

```diff
@@ -184,3 +185,17 @@ test("url with number in subdomain", async () => {
         ".o-mail-Message a:contains(https://www.45017478-master-all.runbot134.odoo.com/web)"
     );
 });
+
+test("isSequential doesn't execute intermediate call.", async () => {
+    const sequential = useSequential();
+    let index = 0;
+    const sequence = () => {
+        index++;
+        const i = index;
+        return sequential(async () => {
+            return new Promise((r) => setTimeout(() => r(i), 1));
```

Comment:
There is one last thing that is missing, it's currently covering that intermediate call don't return any value, but it's not covered whether the inner code is called or not.

This could be fixed with either keeping a step inside id, or incrementing a counter.

Reply:
Should be good now :)